### PR TITLE
SOLR-17467: Start solr in Cloud mode, introduce --user-managed switch for User Managed mode

### PR DESCRIPTION
**Pull request by epugh**
https://issues.apache.org/jira/browse/SOLR-17467

# Description
SolrCloud mode by default!

# Solution

Change default start mode, add `--user-managed` switch.   Update LOTS of bats tests ;-)

# Tests
Bats

# Checklist

Please review the following and check all that apply:

- [ ] I have reviewed the guidelines for [How to Contribute](https://github.com/apache/solr/blob/main/CONTRIBUTING.md) and my code conforms to the standards described there to the best of my ability.
- [ ] I have created a Jira issue and added the issue ID to my pull request title.
- [ ] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended, not available for branches on forks living under an organisation)
- [ ] I have developed this patch against the `main` branch.
- [ ] I have run `./gradlew check`.
- [ ] I have added tests for my changes.
- [ ] I have added documentation for the [Reference Guide](https://github.com/apache/solr/tree/main/solr/solr-ref-guide)

### CODE REVIEW

**Comment by HoustonPutman** [comment]
I haven't done a full review, but it looks good to me so far! We should definitely take a look at the docker scripts to see what we can get rid of...

**Comment by epugh** [comment]
Some discussions on mailing list are leading us to `--classic` as the flag.

**Comment by epugh** [comment]
I realized as I started thinking about documenting `--classic` that it was muddying the waters and adding YAC (_Yet Another Concept_) to the mix.   WHile we may not love the term "user managed" it is whats in the ref guide, and honestly if we are successful, it will just die away in Solr 11, and so then we can deal with it then.  The important thing, the focus of this PR is to change the default start script to Cloud mode.

**Comment by epugh** [comment]
@dsmiley I tagged you on this because I know you don't love the --user-managed, but I think it's the easiest way to get it to done...

**Comment by epugh** [comment]
Just a reminder this PR will only go on `main`.

File: solr/bin/solr [hunk 1]
```diff
@@ -367,14 +367,13 @@ function print_usage() {
...
+    echo "Usage: solr $CMD [-f] [--class] [--host host] [-p port] [-d directory] [-z zkHost] [-m memory] [-e example] [--solr-home solr.solr.home] [--data-home solr.data.home] [--jvm-opts \"jvm-opts\"] [-V]"
     echo ""
     echo "  -f                    Start Solr in foreground; default starts Solr in the background"
     echo "                          and sends stdout / stderr to solr-PORT-console.log"
     echo ""
-    echo "  -c or --cloud         Start Solr in SolrCloud mode; if -z not supplied and ZK_HOST not defined in"
-    echo "                          solr.in.sh, an embedded ZooKeeper instance is started on Solr port+1000,"
-    echo "                          such as 9983 if Solr is bound to 8983"
+    echo "  --user-managed        Start Solr in User Managed mode"
+    echo "                          See the Ref Guide for more details: https://solr.apache.org/guide/solr/latest/deployment-guide/cluster-types.html"
```
Comment (dsmiley) [inline solr/bin/solr:376]: Let's recognize that people cling to the word "standalone" nonetheless, and so the description ought to list whatever aliases people use.  In fact, I might suggest further saying "not SolrCloud".

"Start Solr in User Managed AKA Standalone mode.  This is not SolrCloud, there is no ZooKeeper."
and the ref guide link; sure.

**Comment by malliaridis** [comment]
@epugh FYI this PR seems to have broken `main` for Windows, because I am unable to run Solr in cloud mode under Windows. The output:

```cmd
>bin\solr.cmd start
Java 17 detected. Enabled workaround for SOLR-16463
Unrecognized option: --user-managed
Error: Could not create the Java Virtual Machine.
Error: A fatal exception has occurred. Program will exit.
Waiting up to 30 seconds to see Solr running on port 8983

ERROR: Solr at http://localhost:8983 did not come online within 30 seconds!


ERROR: Solr did not start or was not reachable. Check the logs for errors.
```

When using `bin\solr.cmd start --user-managed` solr starts fine. I am currently investigating the cause.

**Pull request merged**